### PR TITLE
feat(ios): center remote controls and map hardware volume

### PR DESCRIPTION
**Pull request by Quick104**
## Summary

- Center the iOS remote playback controls and improve layout spacing.
- Map hardware volume buttons to remote volume adjustments while suppressing the system HUD.
- Add aspect-ratio controls and rebalance secondary playback actions.

## Testing

- Not run.

## Summary by CodeRabbit

- **New Features**
  - Hardware volume buttons now control remote playback while the app is active, with automatic mute and volume safeguards.
  - Added conditional aspect-ratio controls and refined playback controls, artwork sizing, speaker indicators, and secondary actions.
- **Bug Fixes**
  - Improved server recognition across capitalization, default ports, and equivalent URL formats.
  - Strengthened validation for saved and restored server connections.
- **Tests**
  - Added coverage for server identity matching, URL validation, and edge cases.

### CODE REVIEW

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        if session.outputVolume <= 0.0625 || session.outputVolume >= 0.9375 {
+            nudgeSystemVolume(to: 0.5)
+        }
+    }
+
+    private func nudgeSystemVolume(to value: Float) {
+        suppressUntil = Date().addingTimeInterval(0.5)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) { [weak self] in
+            guard let self else { return }
+            volumeView.subviews.compactMap { $0 as? UISlider }.first?.value = value
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift]: Addressed in 219a5bde: delayed volume writes now capture the MPVolumeView strongly, so restoration runs even after SwiftUI releases the interceptor with the dismissed sheet.

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            Task { @MainActor [weak self] in
+                guard let self, Date() >= suppressUntil,
+                      let old = change.oldValue, let new = change.newValue else { return }
+                if new > old {
+                    onVolumeStep?(1)
+                } else if new < old {
+                    onVolumeStep?(-1)
+                }
+                if new <= 0.0625 || new >= 0.9375 {
+                    nudgeSystemVolume(to: 0.5)
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift]: Addressed in 219a5bde: the interceptor now holds the system volume at a baseline and snaps back after every observed press (own programmatic changes are matched and ignored), so hardware buttons no longer move local audio volume.

File: iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift [hunk 3]
```diff
@@ -61,6 +78,17 @@ struct SiloControlRemoteView: View {
             }
         }
         .preferredColorScheme(.dark)
+        .onAppear {
+            volumeButtons.onVolumeStep = { step in
+                guard let state = controller.state else { return }
+                let current = state.isMuted ? 0 : state.volume
+                let next = min(1, max(0, current + Double(step) / 16))
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift]: Addressed in 219a5bde: added SiloControlClient.stepVolumeOptimistic, which updates state.volume/isMuted optimistically before sending, so rapid presses accumulate instead of resending the same stale absolute value.

File: iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift [hunk 4]
```diff
@@ -61,6 +78,17 @@ struct SiloControlRemoteView: View {
...
+                if step > 0, state.isMuted { controller.setMuted(false) }
+                controller.setVolume(next)
+            }
+            volumeButtons.start()
+        }
+        .onDisappear { volumeButtons.stop() }
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift:88]: Addressed in 219a5bde: the remote view now watches scenePhase and stops interception whenever the scene is not active, restarting only when it returns to active.

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        let claim = claim
+        let canReleaseSession = claim.map {
+            AetherAudioSessionOwnership.canReleaseSharedSession(excluding: $0)
+        } ?? false
+        self.claim = nil
+        if canReleaseSession {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
+                try? AVAudioSession.sharedInstance().setActive(
+                    false,
+                    options: .notifyOthersOnDeactivation
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift]: Addressed in 219a5bde: the delayed closure now re-runs canReleaseSharedSession(excluding:) at execution time. The captured claim stays registered until then but its probe reports inactive, so it blocks nobody while still excluding itself; a restarted remote registers a new active claim and vetoes the deactivation.

File: iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift]: Addressed in 219a5bde: interception now starts only when controller.state is non-nil (and the scene is active), and stops again if the state is lost, via an onChange on controller.state == nil.

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        try? session.setActive(true)
+
+        claim = AetherAudioSessionOwnership.Claim(isHoldingAudio: { [weak self] in
+            self?.isActive ?? false
+        })
+
+        observation = session.observe(\.outputVolume, options: [.old, .new]) { [weak self] _, change in
+            Task { @MainActor [weak self] in
+                guard let self, Date() >= suppressUntil,
+                      let old = change.oldValue, let new = change.newValue else { return }
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift]: Addressed in 219a5bde: the main-actor KVO task now guards on isActive, so observations enqueued before stop() are dropped instead of sending a command after teardown.

File: iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift [hunk 7]
```diff
@@ -61,11 +79,31 @@ struct SiloControlRemoteView: View {
...
         .task(id: controller.state?.contentId) {
             await artwork.resolve(contentId: controller.state?.contentId)
         }
     }
 
+    /// Hardware-volume interception is live only while the scene is active and
+    /// a usable remote state exists: an inactive scene must return the buttons
+    /// to the phone, and without state a press could not reach the TV anyway.
+    private func updateVolumeInterception() {
+        if scenePhase == .active, controller.state != nil {
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/SiloControlRemoteView.swift]: Addressed in ccb3a55f: the gate is now non-empty contentId rather than non-nil state, matching the client's existing idle convention ((contentId ?? "").isEmpty), so a connected-but-idle TV no longer takes over the phone's volume buttons.

File: iosApp/iosApp/Control/iOS/SiloControlClient.swift [hunk 8]
```diff
@@ -368,6 +370,22 @@ final class SiloControlClient {
     func setVolume(_ v: Double) { send(.setVolume(min(max(v, 0), 1))) }
     func setMuted(_ m: Bool) { send(.setMuted(m)) }
 
+    /// Applies one hardware-button volume step. Updates `state` optimistically
+    /// so rapid presses accumulate instead of re-sending the same absolute
+    /// value while the TV's state acknowledgement is still in flight.
+    func stepVolumeOptimistic(_ step: Int) {
+        guard var s = state else { return }
+        let current = s.isMuted ? 0 : s.volume
+        let next = min(1, max(0, current + Double(step) / 16))
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/SiloControlClient.swift]: Addressed in ccb3a55f: steps now start from the retained s.volume rather than the muted display zero, so applyUserVolume never overwrites userVolume with 0. A step down while muted is a no-op (already silent, and the only effect would be destroying the stored level); a step up unmutes and resumes from the retained level.

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    private func setSystemVolume(to value: Float) {
+        pendingProgrammaticVolume = value
+        // Strong capture: restoration on stop() must run even after SwiftUI has
+        // released this interceptor along with the dismissed sheet.
+        let volumeView = volumeView
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            volumeView.subviews.compactMap { $0 as? UISlider }.first?.value = value
+        }
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift:97]: Addressed in f6e4dad2: pendingProgrammaticVolume is now set inside the delayed closure, immediately before the slider write, and only when a slider exists and the session's outputVolume actually differs from the target. The up-then-down-within-100ms case now registers as a real press.

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        // rails so presses in both directions keep producing deltas.
+        baselineVolume = min(max(session.outputVolume, 0.0625), 0.9375)
+
+        observation = session.observe(\.outputVolume, options: [.old, .new]) { [weak self] _, change in
+            Task { @MainActor [weak self] in
+                guard let self, isActive,
+                      let old = change.oldValue, let new = change.newValue else { return }
+                if let pending = pendingProgrammaticVolume, abs(new - pending) < 0.001 {
+                    pendingProgrammaticVolume = nil
+                    return
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift:61]: Addressed in f6e4dad2 (same fix as the CodeRabbit thread on setSystemVolume): the value is claimed at write time rather than schedule time, and only when the write will actually move the volume, so a reversal landing on the pending value is no longer misread as our own write.

File: iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    func start() {
+        guard !isActive else { return }
+        isActive = true
+        pendingProgrammaticVolume = nil
+
+        let session = AVAudioSession.sharedInstance()
+        originalSystemVolume = session.outputVolume
+        if session.category != .playback {
+            try? session.setCategory(.playback, mode: .default, options: [.mixWithOthers])
+        }
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/RemoteHardwareVolumeInterceptor.swift:44]: Addressed in f6e4dad2: the check is now category != .playback OR missing .mixWithOthers. In the second case the existing mode and options are preserved and only .mixWithOthers is added, so a session left non-mixing by a stopped Aether engine no longer interrupts other apps when the remote activates it.

File: iosApp/iosApp/Control/iOS/SiloControlClient.swift [hunk 12]
```diff
@@ -368,6 +370,27 @@ final class SiloControlClient {
...
+    func stepVolumeOptimistic(_ step: Int) {
+        guard var s = state, !(s.isMuted && step < 0) else { return }
+        let next = min(1, max(0, s.volume + Double(step) / 16))
+        if s.isMuted {
+            s.isMuted = false
+            send(.setMuted(false))
+        }
+        s.volume = next
+        state = s
+        send(.setVolume(next))
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Control/iOS/SiloControlClient.swift]: Addressed in f6e4dad2: extracted RemoteVolumeReconciler, which holds the locally-requested volume until the TV echoes it back (or a 4s window lapses, so a dropped command or a change made on the TV isn't held forever). Inbound state is reconciled through it before assignment, and an explicit mute clears the hold. Covered by three new tests, including the two-presses-then-stale-reply case.
